test/pageObjects/ccd-components/fields/ccdAddressUKField.js

```diff
@@ -1,17 +1,38 @@
 class CCDAddressUKField {
 
   constructor(css, id) {
+
     this.css = css;
     let selectorPrefix =  typeof id === 'undefined' ? `${css}` : `${css} div[id="${id}"]`;
+
     this.button = new Button(selectorPrefix + ' .manual-link', 't enter a UK postcode');
+    // this.addressLine1 = new TextField(selectorPrefix + ` input[id*='AddressLine1']`);
+
+
+    this.postcodeText = new TextField('.postcodeinput');
+    this.findAdressButton = new Button('.button', 'Find address');
+    this.addressListDropDown = new DropDown('ccd-write-address-field select');
+    this.addressLine1 = new TextField('#AddressUKField__AddressLine1');
+    this.addressLine2 = new TextField('#AddressUKField__AddressLine2');
+    this.postTown = new TextField('#AddressUKField__PostTown');
+  }
+
+  async isAlertDisplayed() {
+    const EC = protractor.ExpectedConditions;
+    let DEFAULT_TIMEOUT = 10000;
 
-    this.addressLine1 = new TextField(selectorPrefix + ` input[id*='AddressLine1']`);
-    // this.addressLine2 = "";
-    // this.addressLine3 = "";
-    // this.postTown = "";
-    // this.county = "";
-    // this.postcode = "";
-    // this.country = "";
+    try {
+      debugger;
+      await browser.waitForAngular;
+      console.log(browser.getSession());
+      let element = element(by.xpath(`//ccd-alert/cut-alert//div[contains(text(), 'postcode')]`));
+      await browser.wait(EC.visibilityOf(element1), DEFAULT_TIMEOUT)  ;
+      await browser.waitForAngular;
+      return true;
+    } catch (e) {
```

Comment:
would be better to return false instead of throwing an error as function suggests is will return a boolean, then we can use the true false in an assertion in the test

Reply:
👍